x-pack/plugins/ml/public/application/datavisualizer/index_based/index_data_visualizer.tsx

```diff
@@ -189,10 +190,36 @@ export const IndexDataVisualizerPage: FC = () => {
               id="xpack.ml.dataVisualizer.pageHeader"
               defaultMessage="Data Visualizer"
             />
+            {esql ? (
+              <>
+                {` (${i18n.translate('xpack.ml.dataVisualizer.esql', {
+                  defaultMessage: 'ES|QL',
+                })})`}
+                <EuiBetaBadge
```

Comment:
Nit - might be better with some padding between the title and the `Technical preview` badge as we have in e.g. Log pattern analysis.